src/cljs/main/broadfcui/page/workspaces_list.cljs

```diff
@@ -83,31 +83,46 @@
   (->> workspaces (map func) (map count) (apply max)))
 
 (def ^:private access-types ["Project Owner" "Owner" "Writer" "Reader" "No Access"])
+(def ^:private access-types-defaults [true true true true true])
 (def ^:private access-predicates
   {"Project Owner" #(= "PROJECT_OWNER" (% "accessLevel"))
    "Owner" #(= "OWNER" (% "accessLevel"))
    "Writer" #(= "WRITER" (% "accessLevel"))
    "Reader" #(= "READER" (% "accessLevel"))
-   "No Access" #(= "NO ACCESS" (% "accessLevel"))
-   })
+   "No Access" #(= "NO ACCESS" (% "accessLevel"))})
 
-(def ^:private realm-types ["non-TCGA" "TCGA Open Access" "TCGA Protected Access"])
+(def ^:private dataset-types ["Un-published" "Published"])
+(def ^:private dataset-types-defaults [true false])
+(def ^:private dataset-predicates
+  (let [published? #(get-in % ["workspace" "attributes" "library:published"])]
+    {"Published" #(published? %)
+     "Un-published" #(not (published? %))}))
+
+(def ^:private realm-types ["TCGA Open Access" "TCGA Protected Access"])
+(def ^:private realm-types-defaults [false false])
 (def ^:private realm-predicates
- {"non-TCGA" #(not= (config/tcga-namespace) (get-in % ["workspace" "namespace"]))
-  "TCGA Open Access" #(and (= (config/tcga-namespace) (get-in % ["workspace" "namespace"]))
-                           (not (get-in % ["workspace" "realm"])))
-  "TCGA Protected Access" #(and (= (config/tcga-namespace) (get-in % ["workspace" "namespace"]))
-                                (get-in % ["workspace" "realm"]))
-  })
+  (let [tcga? #(= (config/tcga-namespace) (get-in % ["workspace" "namespace"]))
+        inRealm? #(get-in % ["workspace" "realm"])]
+    {"TCGA Open Access" #(and (tcga? %) (not (inRealm? %)))
+     "TCGA Protected Access" #(and (tcga? %) (inRealm? %))
+     ; this pred is used when neither visible option is selected to allow non TCGA workspaces to be shown
+     ; it is not intended to be displayed as an option
+     "TCGA None" #(not (tcga? %))}))
```

Comment:
Similar things here.  We can use `every-pred` to do the `and`s.  So the TCGA Open Access one can be `(every-pred tcga? (complement inRealm?))`